File: app/src/main/java/cc/wenmin92/androidlearn/view/viewpager/ListFragment.java
package cc.wenmin92.androidlearn.view.viewpager;


import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import org.jetbrains.annotations.NotNull;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import cc.wenmin92.androidlearn.R;

public class ListFragment extends Fragment {
    private static final String ARG_PARAM1 = "param1";

    private int mParam;


    public ListFragment() {
    }

    static ListFragment newInstance(int number) {
        ListFragment fragment = new ListFragment();
        Bundle args = new Bundle();
        args.putInt(ARG_PARAM1, number);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam = getArguments().getInt(ARG_PARAM1);
        }
    }

    @Override
    public View onCreateView(@NotNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_list, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        RecyclerView rvList = view.findViewById(R.id.rv_list);
        rvList.setLayoutManager(new LinearLayoutManager(requireContext()));
        rvList.setAdapter(new MyAdapter(mParam));
    }

    static class MyAdapter extends RecyclerView.Adapter<MyViewHolder> {

        private int mParam;

        MyAdapter(int param) {
            mParam = param;
        }

        @NonNull
        @Override
        public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
            return new MyViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_rv, parent, false));
        }

        @SuppressLint("SetTextI18n")
        @Override
        public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
            holder.tvItem.setText(String.format("%s - %s", mParam, position));
        }

        @Override
        public int getItemCount() {
            return 100;
        }
    }

    static class MyViewHolder extends RecyclerView.ViewHolder {

        final TextView tvItem;

        MyViewHolder(@NonNull View itemView) {
            super(itemView);
            tvItem = itemView.findViewById(R.id.tv_item);
        }
    }
}

File: app/src/main/java/cc/wenmin92/androidlearn/view/viewpager/ListFragment.kt
package cc.wenmin92.androidlearn.view.viewpager

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import cc.wenmin92.androidlearn.R

class ListFragment : Fragment() {

    private var mParam: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            mParam = arguments!!.getInt(ARG_PARAM1)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_list, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val rvList = view.findViewById<RecyclerView>(R.id.rv_list)
        rvList.layoutManager = LinearLayoutManager(requireContext())
        rvList.adapter = MyAdapter(mParam)
    }

    internal class MyAdapter(private val mParam: Int) : RecyclerView.Adapter<MyViewHolder>() {

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
            return MyViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_rv, parent, false))
        }

        @SuppressLint("SetTextI18n")
        override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
            holder.tvItem.text = String.format("%s - %s", mParam, position)
        }

        override fun getItemCount(): Int {
            return 100
        }
    }

    internal class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val tvItem: TextView

        init {
            tvItem = itemView.findViewById(R.id.tv_item)
        }
    }

    companion object {
        private val ARG_PARAM1 = "param1"

        internal fun newInstance(number: Int): ListFragment {
            val fragment = ListFragment()
            val args = Bundle()
            args.putInt(ARG_PARAM1, number)
            fragment.arguments = args
            return fragment
        }
    }
}
